Makefile

```diff
@@ -404,10 +407,22 @@ test-gc: all test/gc/build/Release/binding.node
 test-gc-clean:
 	$(RM) -r test/gc/build
 
+ifeq ($(OSTYPE)$(NODE_TARGET_TYPE),darwinshared_library)
+test-build: | all change-install-name build-addons build-addons-napi
+else
 test-build: | all build-addons build-addons-napi
+endif
 
 test-build-addons-napi: all build-addons-napi
 
+# For testing purpose, change the dependency of the executable to point
+# to the shared library with abs path.
+.PHONY: change-install-name
+change-install-name:
+	install_name_tool -change /usr/local/lib/libnode.$(NODE_MODULE_VERSION).dylib \
```

Comment:
Is `/usr/local/lib` hardcoded somewhere else, or should we use the actual prefix here?

Reply:
It's a good question. I don't know where it comes from. and that's important for shared lib embedders. If they build the shared lib, they may want to change it. And that's for macOS only.

Reply:
@yhwang I guess it’s coming from `$(PREFIX)` or `./configure --prefix=…`?

Reply:
Oh, I see. if that's the case, let me change it to variable. Thanks!

Reply:
@addaleax oops, seems the `/usr/local/lib` comes from here: https://github.com/nodejs/node/blob/master/tools/gyp/pylib/gyp/xcode_emulation.py#L763
And it doesn't honor `PREFIX`..... is this a bug?

Reply:
In Linux we use `$ORIGIN` in rpath when linking. For macOS, we can use `@loader_path`. Should I add a TODO in the comment and do the change in other PR?
In here, I will leave it to `/usr/local/lib` for now. is that okay?

Reply:
@addaleax I updated my change to use `@rpath`. Then I reverted my change in `Makefile`. You may take a look.